client/gutenboarding/store/resolvers.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import * as selectors from './selectors';
+import { apiFetch, dispatch } from '@wordpress/data-controls';
+import { STORE_KEY } from './constants';
+
+type ResolverParameters< S extends ( state: any, ...args: any ) => any > = S extends (
+	state: any,
+	...args: infer P
+) => any
+	? P
+	: never;
+
+/**
+ * Requests theme supports data from the index.
+ */
+export function* getVertical(
+	...[ search ]: ResolverParameters< typeof selectors[ 'getVertical' ] >
```

Comment:
I don't quite get why we're wrapping `search` in an array only to destructure it again?

Reply:
Just being silly 😛 This has been improved.